python/pyspark/taskcontext.py

```diff
@@ -193,17 +196,16 @@ def _initialize(cls, port, secret):
 
     def barrier(self):
         """
-        .. note:: Experimental
-
         Sets a global barrier and waits until all tasks in this stage hit this barrier.
         Similar to `MPI_Barrier` function in MPI, this function blocks until all tasks
         in the same stage have reached this routine.
 
+        .. versionadded:: 2.4.0
+        .. note:: Experimental
```

Comment:
Just thinking out loud ‒ shall we differentiate experimental APIs? Let's say with `.. caution::` instead of `.. note::`?

Reply:
I converted `Evolving` to `This API is evolving` and place under `Notes` for now in the previous SQL PR, but sure I don't think what I did is the best way. `.. caution::` sounds fine to me too if the output looks better.

These `Evolving`, `Experimental`, etc are from [the tags on Scala API side](https://github.com/apache/spark/tree/master/common/tags/src/main/java/org/apache/spark/annotation).